Detectors/Calibration/include/DetectorsCalibration/MeanVertexCalibrator.h

```diff
@@ -0,0 +1,112 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef MEAN_VERTEX_CALIBRATOR_H_
+#define MEAN_VERTEX_CALIBRATOR_H_
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DetectorsCalibration/MeanVertexData.h"
+#include "DataFormatsCalibration/MeanVertexObject.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include <array>
+#include <deque>
+
+namespace o2
+{
+namespace calibration
+{
+
+class MeanVertexCalibrator final : public o2::calibration::TimeSlotCalibration<o2::dataformats::PrimaryVertex, o2::calibration::MeanVertexData>
+{
+  using PVertex = o2::dataformats::PrimaryVertex;
+  using MeanVertexData = o2::calibration::MeanVertexData;
+  using TFType = uint64_t;
+  using Slot = o2::calibration::TimeSlot<MeanVertexData>;
+  using MVObject = o2::dataformats::MeanVertexObject;
+  using MVObjectVector = std::vector<MVObject>;
+  using CcdbObjectInfo = o2::ccdb::CcdbObjectInfo;
+  using CcdbObjectInfoVector = std::vector<CcdbObjectInfo>;
+
+ public:
+ MeanVertexCalibrator(int minEnt = 500, bool useFit = false, int nBinsX = 1000, float rangeX = 10.f,
+		      int nBinsY = 1000, float rangeY = 10.f, int nBinsZ = 1000, float rangeZ = 10.f,
+		      int nSlotsSMA = 5) :
```

Comment:
These values are not to be considered, because it is the device that sets them. But I can change it to avoid confusion. 
The units are cm, right? At least from what I saw (I think I did not see in the PrimaryVertex and Vertex class). From the simulated events, 0.1 was enough for X and Y.

Reply:
Updated, also the binning (as I said, the device is set differently, but better have meaningful ranges and bins).

Reply:
yes, all units are cm. But in the constructor I still see old defaults.

Reply:
Done. Sorry that I had forgotten.